apps/asap-server/src/handlers/webhooks/webhook-events-updated.ts

```diff
@@ -1,11 +1,105 @@
-import { APIGatewayProxyHandlerV2 } from 'aws-lambda';
-import debug from 'debug';
+/* eslint-disable no-shadow */
+import { calendar_v3 as calendarV3, Auth } from 'googleapis';
+import { framework as lambda } from '@asap-hub/services-common';
 
-const logger = debug('asap-server');
+import { Handler } from '../../utils/types';
+import { http } from '../../utils/instrumented-framework';
+import Calendars, { CalendarController } from '../../controllers/calendars';
+import {
+  syncCalendarFactory,
+  SyncCalendarFactory,
+} from '../../utils/sync-google-calendar';
+import getJWTCredentials, {
+  GetJWTCredentials,
+} from '../../utils/aws-secret-manager';
+import logger from '../../utils/logger';
 
-export const handler: APIGatewayProxyHandlerV2<void> = async (
-  event,
-  context,
-) => {
-  logger('Request received', JSON.stringify({ event, context }, null, 2));
+export const webhookEventUpdatedHandlerFactory = (
+  calendars: CalendarController,
+  getJWTCredentials: GetJWTCredentials,
+  syncCalendarFactory: SyncCalendarFactory,
+): Handler =>
+  http(
+    async (request: lambda.Request): Promise<lambda.Response> => {
+      // TODO: validate google request auth
+
+      const resourceId = request.headers['x-goog-resource-id'];
+      if (!resourceId) {
+        return {
+          statusCode: 400,
+          payload: {
+            message: 'Missing x-goog-resource-id header',
+          },
+        };
+      }
+
+      const calendar = await calendars
+        .fetchByResouceId(resourceId)
+        .catch((err) => {
+          logger('Error fetching calendar', err);
+          return undefined;
+        });
+
+      if (!calendar?.id) {
+        return { statusCode: 502 };
+      }
+      const squidexCalendarId = calendar.id;
+      const googleCalendarId = calendar.data.id.iv;
+
+      const syncToken = await calendars
```

Comment:
why not just get it from 
```typescript
const syncToken = calendar.data.syncToken
```
?

Reply:
Just... Thanks. 🤦🏽‍♂️  
I guess, copy and pasting old versions of your code is not the best strategy 😂